suites/e2e/suite.js

```diff
@@ -43,11 +47,22 @@ module.exports = {
 			delete this.suite.options.balenaOS.network.wireless;
 		}
 
+// Downloads the balenaOS image that will be flashed to the DUT 
+// This is optional, you can provide your own balenaOS images as well. 
```

Comment:
If the user does provide their own image, does this fetching of an image from the api happen anyway?

Reply:
It does. I will put an IF statement to stop it if it is `!false`

Reply:
@rcooke-warwick We don't pass the `image` property to the tests for it to ever find this out. It can't happen until we start passing the whole config.json to the tests. 

What we do have available for options is `this.suite.options` which is basically this file:

```

// Device identification
function uid(a) {
	return a
		? (a ^ (Math.random() * 16)).toString(16)
		: ([1e15] + 1e15).replace(/[01]/g, uid);
}
// Test identification
const id = `${Math.random()
	.toString(36)
	.substring(2, 10)}`;

module.exports = options => {
	return {
		id,
		balena: {
			apiUrl: options.balenaApiUrl,
		},
		balenaOS: {
			config: {
				uuid: uid(),
				pubKey: options.osPubkey,
			},
			download: {
				type: options.downloadType,
				version: options.downloadVersion,
				source: options.downloadSource,
			},
			network: {
				wired: options.networkWired,
				wireless: options.networkWireless,
			},
			releaseInfo: {
				version: options.osReleaseVersion,
				variant: options.osReleaseVariant,
			},
		},
	};
};
```

Let's also talk about why suites have this additional conf.js file?

Reply:
https://github.com/balena-os/leviathan/issues/600

Reply:
@vipulgupta2048 if I understand correctly, the problem here is that in the test suite, we want to know if an image was actually provided - if it wasn't, then we fetchOS. However the test suite doesn't know if it was sent an image or not, because it doesn't see the complete `config.js`?

Reply:
About the additional `conf.js` file - it does feel weird doesn't it - but I think its there for configuration information generated at runtime?

Reply:
Although many parameters are just assigned from the external `config.js` anyway... I;m not sure why this intermediate step is there